src/test/IsLessThanOrEqualToSumOf.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { expect } from 'chai'
+import { IsLessThanOrEqualToSumOf, IsLessThanOrEqualToSumOfConstraint } from '../main/isLessThanOrEqualToSumOf'
+import { validateSync, ValidationArguments } from 'class-validator'
+
+const DEFAULT_SUM = 30
+const DEFAULT_VALUE = 200
+
+class LessThanOrEqualToSumTest {
+  @IsLessThanOrEqualToSumOf('defaultSum', 'targetValue')
+  value: any
+  targetValue: number
+  defaultSum: number
+
+  constructor (value: any, sum: any, lessThan: any) {
+    this.value = value
+    this.defaultSum = sum
+    this.targetValue = lessThan
+  }
+}
+
+describe('IsLessThanOrEqualToSumOf', () => {
+
+  describe('validate', () => {
+
+    describe('should return true when ', () => {
+      it('given an undefined value', () => {
+        expect(validateSync(new LessThanOrEqualToSumTest(undefined, DEFAULT_SUM, DEFAULT_VALUE))).to.be.length(0)
```

Comment:
Having `to.be.empty` would be more readable for me.

In long term I think we could have our own validation assertions to be able to say for example:

```
expect(validate(new LessThanOrEqualToSumTest(undefined, DEFAULT_SUM, DEFAULT_VALUE))).to.be.valid

expect(validate(new LessThanOrEqualToSumTest(undefined, DEFAULT_SUM, DEFAULT_VALUE))).to.reportError('Invalid xyz')
```